compiler-plugin/src/main/java/io/ballerina/stdlib/jwt/compiler/staticcodeanalyzer/JWTCipherAlgorithmAnalyzer.java

```diff
@@ -0,0 +1,190 @@
+import io.ballerina.compiler.syntax.tree.MappingFieldNode;
+import io.ballerina.compiler.syntax.tree.ModuleMemberDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.ModuleVariableDeclarationNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.PositionalArgumentNode;
+import io.ballerina.compiler.syntax.tree.QualifiedNameReferenceNode;
+import io.ballerina.compiler.syntax.tree.SimpleNameReferenceNode;
+import io.ballerina.compiler.syntax.tree.SpecificFieldNode;
+import io.ballerina.compiler.syntax.tree.StatementNode;
+import io.ballerina.compiler.syntax.tree.VariableDeclarationNode;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.plugins.AnalysisTask;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.scan.Reporter;
+
+/**
+ * Analyzes JWT cipher algorithm usage in Ballerina code to detect insecure configurations.
+ * <p>
+ * This analysis task inspects calls to <code>jwt:issue()</code> and reports if the signature configuration uses
+ * the insecure <code>NONE</code> algorithm. It checks for insecure usage in inline mapping literals, function-local
+ * variables, and module-level variables.
+ * </p>
+ */
+public class JWTCipherAlgorithmAnalyzer implements AnalysisTask<SyntaxNodeAnalysisContext> {
+    private final Reporter reporter;
+    private static final String JWT = "jwt";
+    private static final String ISSUE = "issue";
+    private static final String SIGNATURE_CONFIG = "signatureConfig";
+    private static final String ALGORITHM = "algorithm";
+    private static final String NONE = "NONE";
+
+    public JWTCipherAlgorithmAnalyzer(Reporter reporter) {
+        this.reporter = reporter;
+    }
+
+    /**
+     * Analyzes jwt:issue() calls and reports usage of signatureConfig.algorithm == NONE.
+     * It checks inline mappings, function-local variables, and module-level variables.
+     *
+     * @param context the syntax node analysis context for reporting issues
+     */
+    @Override
+    public void perform(SyntaxNodeAnalysisContext context) {
+        FunctionCallExpressionNode functionCall = (FunctionCallExpressionNode) context.node();
+
+        // Check if the function is qualified with the module name
+        if (!(functionCall.functionName() instanceof QualifiedNameReferenceNode qualifiedName)) {
+            return;
+        }
+
+        // Verify the module and function name
+        if (!JWT.equals(qualifiedName.modulePrefix().text()) || !ISSUE.equals(qualifiedName.identifier().text())) {
+            return;
+        }
+
+        for (FunctionArgumentNode arg : functionCall.arguments()) {
+            if (!(arg instanceof PositionalArgumentNode posArg)) {
+                continue;
+            }
+            ExpressionNode expr = posArg.expression();
+            if (expr instanceof MappingConstructorExpressionNode mappingConstructor) {
```

Comment:
The mapping constructor can be part of other kinds of expressions too - e.g., type casts, parenthesized (expr within parenthesis)